bootloader/src/pyi_utils.c

```diff
@@ -220,16 +220,41 @@ pyi_unsetenv(const char *variable)
 
 /* TODO rename fuction and revisit */
 int
-pyi_get_temp_path(char *buffer)
+pyi_get_temp_path(char *buffer, char *runtime_tmpdir)
 {
     int i;
     wchar_t *wchar_ret;
     wchar_t prefix[16];
     wchar_t wchar_buffer[PATH_MAX];
+    size_t requiredSize;
+    wchar_t *original_tmpdir;
+    wchar_t wruntime_tmpdir[PATH_MAX + 1];
+    wchar_t wruntime_tmpdir_abspath[PATH_MAX + 1];
+
+    if (NULL != runtime_tmpdir) {
+      /*
+       * Get original TMP environment variable so it can be restored
+       * after this is done.
+       */
+      _wgetenv_s(&requiredSize, NULL, 0, L"TMP");
+      if (requiredSize == 0) {
+        original_tmpdir = NULL;
+      } else {
+        original_tmpdir = (wchar_t *) malloc(requiredSize * sizeof(wchar_t));
+        if (!original_tmpdir) {
+          return 0;
+        }
+        _wgetenv_s(&requiredSize, original_tmpdir, requiredSize, L"TMP");
+      }
+      /*
+       * Set TMP to runtime_tmpdir for _wtempnam() later
+       */
+      pyi_win32_utils_from_utf8(wruntime_tmpdir, runtime_tmpdir, PATH_MAX);
+      _wfullpath(wruntime_tmpdir_abspath, wruntime_tmpdir, PATH_MAX);
```

Comment:
Please use `pyi_path_fullpath`. If thtat function is incomplete, please update that.